src/PowerShellEditorServices/Console/WindowsConsoleOperations.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices.Console
+{
+    internal class WindowsConsoleOperations : IConsoleOperations
+    {
+        private ConsoleKeyInfo? _bufferedKey;
+
+        private SemaphoreSlim _readKeyHandle = new SemaphoreSlim(1, 1);
+
+        public async Task<ConsoleKeyInfo> ReadKeyAsync(CancellationToken cancellationToken)
+        {
+            await _readKeyHandle.WaitAsync(cancellationToken);
+            try
+            {
+                return
+                    _bufferedKey.HasValue
```

Comment:
Can this ever be anything but `false`?  `_bufferedKey` is private and I'm not seeing how it would get set before we reach this point.

Reply:
@rkeithhill it's set on line 22.  I was trying to avoid the whole [temp variable dance](https://github.com/PowerShell/PowerShellEditorServices/blob/764db9427eb7418ac3b553de0404177cb50f20c5/src/PowerShellEditorServices/Console/ConsoleReadLine.cs#L485) from the original implementation.  Might have went a little too complicated with it though.  @daviwil thoughts?